Software/Development/SemesterProject/App/highPriorityTasks.c

```diff
@@ -26,10 +18,26 @@ __interrupt void adc_isr(void)
 {
     readHighPrioritySignals();
 
-    AdcRegs.ADCINTFLGCLR.bit.ADCINT1 = 1;
-    PieCtrlRegs.PIEACK.all = PIEACK_GROUP1;   // Acknowledge interrupt to PIE
+    acknowledgeAdcInterrupt();
 
     executeHighPriorityTasks();
 

```

Comment:
Im not 100% percent convinced about readHighPrioritySignals not being part of executeHighPriorityTasks... Seems like read signals should be at the same level as read position or perform control. No big deal anyway

Reply:
The idea was to be able to leave the interrupt after reading the signals, if a new interrupt had been received. And that is why the acknowledge is in between them